photon-client/src/main/scala/com/linkedin/photon/ml/cli/game/GameDriver.scala

```diff
@@ -14,27 +14,149 @@
  */
 package com.linkedin.photon.ml.cli.game
 
-import org.apache.hadoop.conf.Configuration
+import org.apache.commons.cli.MissingArgumentException
+import org.apache.hadoop.fs.Path
 import org.apache.spark.SparkContext
+import org.apache.spark.ml.param.{Param, ParamValidators, Params}
+import org.apache.spark.ml.util.Identifiable
 import org.slf4j.Logger
 
+import com.linkedin.photon.ml.DataValidationType.DataValidationType
+import com.linkedin.photon.ml.Types.FeatureShardId
+import com.linkedin.photon.ml.data.InputColumnsNames
 import com.linkedin.photon.ml.data.avro.NameAndTermFeatureSetContainer
+import com.linkedin.photon.ml.evaluation.EvaluatorType
+import com.linkedin.photon.ml.io.FeatureShardConfiguration
 import com.linkedin.photon.ml.util.Implicits._
 import com.linkedin.photon.ml.util._
 
 /**
- * Contains common functions for GAME training and scoring drivers.
- *
- * @param sc The Spark context for the application
- * @param params The input parameters
- * @param logger The logger instance for recording log output
+ * Contains common parameters and functions for GAME training/scoring drivers.
  */
-abstract class GameDriver(sc: SparkContext, params: FeatureParams with PalDBIndexMapParams, logger: Logger) {
+trait GameDriver extends Params {
```

Comment:
@joshvfleming 
@fastier-li 
@Adama94

`GameDriver` now defines the parameters that are shared by the training and scoring drivers, in addition to the functions shared by the drivers.